packages/back-end/src/enterprise/billing/index.ts

```diff
@@ -45,13 +46,46 @@ export async function deletePaymentMethodById(
   paymentMethodId: string
 ) {
   const url = `${LICENSE_SERVER_URL}subscription/payment-methods/detach`;
-  const res = await callLicenseServer(
+  const res = await callLicenseServer({
     url,
-    JSON.stringify({
+    body: JSON.stringify({
       licenseKey,
       paymentMethodId,
       cloudSecret: process.env.CLOUD_SECRET,
-    })
-  );
+    }),
+  });
   return res;
 }
+
+export async function getUsageDataFromServer(
+  organization: string
+): Promise<OrganizationUsage> {
+  const url = `${LICENSE_SERVER_URL}cdn/${organization}/usage`;
+
+  const usage = await callLicenseServer({ url, method: "GET" });
+
+  return {
+    ...usage,
+    cdn: { ...usage.cdn, lastUpdated: new Date(usage.cdn.lastUpdated) },
+  };
+}
+
+const keyToUsageData: Record<string, OrganizationUsage> = {};
+
+export async function getUsage(organization: string) {
+  const cacheCutOff = new Date();
+  cacheCutOff.setHours(cacheCutOff.getHours() - 1);
+
+  Object.keys(keyToUsageData).forEach((organization) => {
+    if (keyToUsageData[organization]?.cdn.lastUpdated <= cacheCutOff)
+      delete keyToUsageData[organization];
+  });
+
+  if (keyToUsageData[organization]) return keyToUsageData[organization];
+
+  const usage = await getUsageDataFromServer(organization);
```

Comment:
We need to catch exceptions here and handle them.  Otherwise, this will be an uncaught promise rejection, which can crash the entire Node process.